ros2bag/ros2bag/verb/record.py

```diff
@@ -48,10 +48,12 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             help='disables topic auto discovery during recording: only topics present at '
              'startup will be recorded')
         parser.add_argument(
-            '-p', '--polling-interval', default=100,
+            '-p', '--polling-interval', type=int, default=100,
             help='time in ms to wait between querying available topics for recording. It has no '
              'effect if --no-discovery is enabled.'
         )
+        self._subparser = parser
```

Comment:
You don't need that line.

Reply:
@Karsten1987  but without it won't line number 92 throw error ? ( inside the else statement which calls             `self._subparser.print_help()` )

Reply:
you're absolutely right!